test/controllers/EtmpCheckControllerTest.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import models.EtmpRegistrationDetails
+import org.mockito.Matchers
+import org.mockito.Mockito.when
+import org.scalatest.mockito.MockitoSugar
+import play.api.libs.json.Json
+import play.api.mvc.ControllerComponents
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import services.EtmpRegimeService
+import utils.{BaseSpec, AwrsTestJson}
+
+import scala.concurrent.Future
+
+class EtmpCheckControllerTest extends BaseSpec with MockitoSugar {
+
+  val mockEtmpRegimeService: EtmpRegimeService = mock[EtmpRegimeService]
+  val cc: ControllerComponents = app.injector.instanceOf[ControllerComponents]
+
+  object TestEtmpCheckController extends EtmpCheckController(cc, mockEtmpRegimeService)
+
+  "checkEtmp" should {
+    "return an OK" when {
+      "there is a regime model and there is ETMP registration details for an organisation" in {
+        val etmpRegistrationDetails = EtmpRegistrationDetails(
```

Comment:
GREY - For readability, I'd normally have all these variables in one line, if I split them up like this, then it's good to have the variable name e.g. organisationName = None. Just clearer to read at a glance.